backend/services/usage_metrics_service.py

```
@@ -0,0 +1,262 @@
+            else:
+                # Handle string format if needed
+                dt = datetime.fromisoformat(str(created_at).replace("Z", "+00:00"))
+                created_at_epoch = dt.timestamp()
+
+            # Calculate time difference in seconds, then convert to ms
+            # timestamp_awaiting_finalization is already in seconds
+            processing_seconds = float(timestamp_awaiting) - created_at_epoch
+            # appeal_processing_time is already in milliseconds
+            total_ms = int(processing_seconds * 1000) + int(appeal_processing_time)
+            return max(0, total_ms)  # Ensure non-negative
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)